packages/ui/src/components/ValidatorList/KeystoresDataGrid.tsx

```diff
@@ -61,6 +69,106 @@ export default function KeystoresDataGrid({
     openDashboardTab();
   }, [validatorSummaryURL]);
 
+  useState<CustomValidatorUpdateRequest>();
+  const [validatorsSubscriptionMap, setValidatorsData] =
+    useState<SmoothStatusByPubkey>({});
+  const [oracleCallError, setOracleCallError] = useState<string>();
+
+  // Check that Smooth API returns an expected response format
+  function isValidOracleResponse(response: SmoothValidatorByIndexApiResponse): boolean {
+    return (
+      response &&
+      (Array.isArray(response.found_validators) ||
+        response.found_validators === null) &&
+      (Array.isArray(response.not_found_validators) ||
+        response.not_found_validators === null)
+    );
+  }
+
+  // This function fetches the subscription status of the validators in the rows array
+  // It uses the validator index to fetch the status from the Oracle API
+  // It updates the validatorsSubscriptionMap state with the new data
+  const fetchValidatorsData = async (rows: CustomValidatorGetResponse[]) => {
+    const newValidatorSubscriptionStatus: SmoothStatusByPubkey = {};
+    // Filter rows to include only those with an index
+    const rowsWithIndex = rows.filter(row => row.index);
```

Comment:
Do you set a specific icon for those validators that have not received a deposit yet? It could be great to have this information given to the user

Reply:
Tooltip: "No index found for this validator. Are your clients synced? Have you done a deposit? "

Reply:
If 0 index are found --> consensus client probably offline. Oracle call returns error, so a red "?" is rendered with error message when hovering. 
https://github.com/dappnode/StakingBrain/blob/marc/smooth-status-integration/packages/ui/src/components/ValidatorList/SmoothStatus.tsx#L65

If only some index could not be found --> their subscriptionStatus is not fetched (does not exist) so they are stuck in "Loading..." I could improve this but its not that bad neither https://github.com/dappnode/StakingBrain/blob/marc/smooth-status-integration/packages/ui/src/components/ValidatorList/SmoothStatus.tsx#L74